launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -392,7 +392,7 @@ boolean isClientMode(Map<String, String> userProps) {
     // Default master is "local[*]", so assume client mode in that case
     return userMaster == null ||
       "client".equals(userDeployMode) ||
-      (!userMaster.equals("yarn") && userDeployMode == null);
+      userDeployMode == null;
```

Comment:
Sorry for the last minute request, but shall we change three lines into one-liner?
```java
-    return userMaster == null ||
-      "client".equals(userDeployMode) ||
-      (!userMaster.equals("yarn") && userDeployMode == null);
+    return userMaster == null || userDeployMode == null || "client".equals(userDeployMode);
```